contracts/interfaces/ICurveRegistry.sol

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.11;
+pragma abicoder v1;
+
+interface ICurveRegistry {
+    function find_pool_for_coins(address _srcToken, address _dstToken) external view returns ( address );
```

Comment:
You have to use `find_pool_for_coins(_from: address, _to: address, i: uint256 = 0)` implementation and calculate rate considering all pools for this pair.